relays/lib-substrate-relay/src/on_demand/headers.rs

```diff
@@ -146,8 +146,13 @@ impl<P: SubstrateFinalitySyncPipeline> OnDemandRelay<P::SourceChain, P::TargetCh
 				finality_source.prove_block_finality(current_required_header).await?;
 			let header_id = header.id();
 
-			// optimize justification before including it into the call
-			P::FinalityEngine::optimize_proof(&self.target_client, &header, &mut proof).await?;
+			// verify and optimize justification before including it into the call
+			let current_set_id = P::FinalityEngine::verify_and_optimize_proof(
```

Comment:
Nit: Is this a leftover ? Shouldn't it be something like  `finality_verification_context` ?